src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -99,7 +99,7 @@ object Parsers {
     /** Issue an error at given offset if beyond last error offset
       *  and update lastErrorOffset.
       */
-    def syntaxError(msg: Message, offset: Int = in.offset): Unit =
+    def syntaxError(msg: => Message, offset: Int = in.offset): Unit =
```

Comment:
Yes, the store reporter will force if debugging - but I think we can live with that :)